mesonbuild/compilers/mixins/arm.py

```diff
@@ -127,30 +133,37 @@ def compute_parameters_with_absolute_paths(self, parameter_list: T.List[str], bu
 
 
 class ArmclangCompiler:
-    def __init__(self):
+
+    if T.TYPE_CHECKING:
+        can_compile_suffixes = set()  # type: T.Set[str]
+        is_cross = True
+        version = '0'
+
+        def get_pch_name(self, name: str) -> str: ...
+
+    def __init__(self) -> None:
         if not self.is_cross:
             raise mesonlib.EnvironmentException('armclang supports only cross-compilation.')
         # Check whether 'armlink' is available in path
         self.linker_exe = 'armlink'
         args = '--vsn'
         try:
-            p, stdo, stderr = mesonlib.Popen_safe(self.linker_exe, args)
+            p, stdo, stderr = mesonlib.Popen_safe([self.linker_exe], args)
         except OSError as e:
             err_msg = 'Unknown linker\nRunning "{0}" gave \n"{1}"'.format(' '.join([self.linker_exe] + [args]), e)
             raise mesonlib.EnvironmentException(err_msg)
         # Verify the armlink version
-        ver_str = re.search('.*Component.*', stdo)
-        if ver_str:
-            ver_str = ver_str.group(0)
+        _ver_str = re.search('.*Component.*', stdo)
+        if _ver_str:
+            ver_str = str(_ver_str.group(0))
         else:
             raise mesonlib.EnvironmentException('armlink version string not found')
-        assert ver_str  # makes mypy happy
         # Using the regular expression from environment.search_version,
         # which is used for searching compiler version
         version_regex = r'(?<!(\d|\.))(\d{1,2}(\.\d+)+(-[a-zA-Z0-9]+)?)'
-        linker_ver = re.search(version_regex, ver_str)
-        if linker_ver:
-            linker_ver = linker_ver.group(0)
+        _linker_ver = re.search(version_regex, ver_str)
+        if _linker_ver:
+            linker_ver = _linker_ver.group(0)
         if not mesonlib.version_compare(self.version, '==' + linker_ver):
```

Comment:
`if not _linker_ver` won't this error out saying `linker_ver` is not defined?

Reply:
Ah, yup. Good catch

Reply:
Actually, this code is all silly, we already have the linker version in self.linker.version... I'm going to just change it to do that.